### PR TITLE
PB-2042: Fixing regression as follows - Don't count job pod count in case of job success

### PR DESCRIPTION
**Pull request by prashanthpx**
**What this PR does / why we need it**:
Fixing regression as follows - Don't count job pod count in case of job success

**Which issue(s) this PR fixes** (optional)
PB-2042

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/controllers/dataexport/reconcile.go [hunk 1]
```diff
@@ -345,24 +345,26 @@ func (c *Controller) sync(ctx context.Context, in *kdmpapi.DataExport) (bool, er
...
-		// can be coming later during which is pod might be deleted. GetPods() doesn't give
-		// error when pods are not present hence we are not sure if pods are still coming up
-		// or deleted.
-		if progress.RestartCount < (utils.JobPodBackOffLimit - 1) {
-			logrus.Tracef("pod restart cnt: %v", progress.RestartCount)
-			data := updateDataExportDetail{
-				status: kdmpapi.DataExportStatusInProgress,
-			}
-			return true, c.updateStatus(dataExport, data)
+			// When job reaches the last retry limit, pod is removed and reconciler
```
Comment (diptiranjanpx) [inline pkg/controllers/dataexport/reconcile.go:348]: We should move this condition check inside the below switch case

`case drivers.JobStateFailed:`

Comment (prashanthpx) [reply pkg/controllers/dataexport/reconcile.go]: As discussed, the case statement is for final failure after all retries have been exhausted.

File: pkg/drivers/utils/utils.go [hunk 2]
```diff
@@ -124,11 +124,15 @@ func FetchJobContainerRestartCount(j *batchv1.Job) (int32, error) {
 	)
 	if err != nil {
 		// Cannot determine job state
-		return 0, fmt.Errorf("cannot determine job state")
+		errMsg := "cannot determine job state"
```
Comment (diptiranjanpx) [inline pkg/drivers/utils/utils.go:127]: put job namespace/job name for better debugging

File: pkg/drivers/utils/utils.go [hunk 3]
```diff
@@ -124,11 +124,15 @@ func FetchJobContainerRestartCount(j *batchv1.Job) (int32, error) {
...
+		logrus.Errorf("%v", errMsg)
+		return 0, fmt.Errorf(errMsg)
 	} else if len(pods.Items) == 0 {
+		logrus.Debugf("no pods in job spec: %v/%v", j.Namespace, j.Name)
 		return 0, nil
 	}
 	if len(pods.Items[0].Status.ContainerStatuses) == 0 {
+		logrus.Debugf("container status not present in pod")
```
Comment (diptiranjanpx) [inline pkg/drivers/utils/utils.go:135]: put job namespace, job and pod name.

File: pkg/controllers/dataexport/reconcile.go [hunk 4]
```diff
@@ -356,11 +358,23 @@ func (c *Controller) sync(ctx context.Context, in *kdmpapi.DataExport) (bool, er
 		// can be coming later during which is pod might be deleted. GetPods() doesn't give
 		// error when pods are not present hence we are not sure if pods are still coming up
 		// or deleted.
+		// TODO: Remove these debug later
+		logrus.Debugf("pod restart cnt: %v", progress.RestartCount)
```
Comment (diptiranjanpx) [inline pkg/controllers/dataexport/reconcile.go:362]: Make all 3 debug as a single line

File: pkg/controllers/dataexport/reconcile.go [hunk 5]
```diff
@@ -356,11 +358,23 @@ func (c *Controller) sync(ctx context.Context, in *kdmpapi.DataExport) (bool, er
...
+		logrus.Debugf("driver state: %v", drivers.JobStateInProgress)
+		logrus.Debugf("CR name: %v, state: %v", dataExport.Name, dataExport.Status.Status)
 		if progress.RestartCount < (utils.JobPodBackOffLimit - 1) {
-			logrus.Tracef("pod restart cnt: %v", progress.RestartCount)
```
Comment (diptiranjanpx) [inline pkg/controllers/dataexport/reconcile.go:360]: put pod name in the log

File: pkg/controllers/dataexport/reconcile.go [hunk 6]
```diff
@@ -356,11 +358,23 @@ func (c *Controller) sync(ctx context.Context, in *kdmpapi.DataExport) (bool, er
...
+			if (progress.State == drivers.JobStateInProgress) || (progress.State == drivers.JobStateFailed) {
+				data := updateDataExportDetail{
+					status: kdmpapi.DataExportStatusInProgress,
+				}
+				logrus.Debugf("Retrying job again internally ...")
```
Comment (diptiranjanpx) [inline pkg/controllers/dataexport/reconcile.go:370]: put job name in the log

File: pkg/controllers/dataexport/reconcile.go [hunk 7]
```diff
@@ -356,11 +358,23 @@ func (c *Controller) sync(ctx context.Context, in *kdmpapi.DataExport) (bool, er
...
+				return true, c.updateStatus(dataExport, data)
+			}
+		} else {
 			data := updateDataExportDetail{
-				status: kdmpapi.DataExportStatusInProgress,
+				status: kdmpapi.DataExportStatusFailed,
 			}
+			logrus.Debugf("Exhausted job retries and job is still failing ...")
```
Comment (diptiranjanpx) [inline pkg/controllers/dataexport/reconcile.go:377]: put job name in the log

File: pkg/controllers/dataexport/reconcile.go (hunk 6 above)
Comment (prashanthpx) [reply pkg/controllers/dataexport/reconcile.go:370]: In this context, we don't have job name and pod.  Would need more code to get the ref here. So keeping it as is.

File: pkg/controllers/dataexport/reconcile.go (hunk 7 above)
Comment (prashanthpx) [reply pkg/controllers/dataexport/reconcile.go:377]: Same as above.

File: pkg/drivers/utils/utils.go (hunk 2 above)
Comment (prashanthpx) [reply pkg/drivers/utils/utils.go:127]: Have added job name, job ns where ever the context is available.

**Review by siva-portworx — APPROVED** [review]

**Pull request merged**